cdap-app-fabric/src/test/java/io/cdap/cdap/master/environment/DefaultMasterEnvironmentRunnableContextTest.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.master.environment;
+
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.common.discovery.ResolvingDiscoverable;
+import io.cdap.cdap.common.discovery.URIScheme;
+import io.cdap.cdap.common.guice.LocalLocationModule;
+import io.cdap.common.http.HttpMethod;
+import io.cdap.http.AbstractHttpHandler;
+import io.cdap.http.HttpResponder;
+import io.cdap.http.NettyHttpService;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import org.apache.twill.common.Cancellable;
+import org.apache.twill.discovery.DiscoveryService;
+import org.apache.twill.discovery.DiscoveryServiceClient;
+import org.apache.twill.discovery.InMemoryDiscoveryService;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.nio.charset.StandardCharsets;
+import java.util.stream.Collectors;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+
+/**
+ * Test for {@link DefaultMasterEnvironmentRunnableContextTest}.
+ */
+public class DefaultMasterEnvironmentRunnableContextTest {
+  private static DefaultMasterEnvironmentRunnableContext context;
+  private static DiscoveryService discoveryService;
+  private static Cancellable cancelDiscovery;
+  private static NettyHttpService httpService;
+
+  @BeforeClass
+  public static void setup() throws Exception {
+    discoveryService = new InMemoryDiscoveryService();
+    Injector injector = Guice.createInjector(
+      new LocalLocationModule(),
+      new AbstractModule() {
+        @Override
+        protected void configure() {
+          bind(CConfiguration.class).toInstance(CConfiguration.create());
+          bind(DiscoveryServiceClient.class).toInstance((DiscoveryServiceClient) discoveryService);
+        }
+      });
+    context = injector.getInstance(DefaultMasterEnvironmentRunnableContext.class);
```

Comment:
Not really need to use Guice. Just new the `LocalLocationFactory` with a base path pointing to a test directory created via `TemporaryFolder` is better.